### PR TITLE
fix: Avoid raising events when recycling TextBox, ToggleSwitch, and ToggleButton

### LINKED ISSUE #12182: [WASM] CheckBox.Unchecked fired ~1 minute *after* it was removed from the visual tree/DOM
### Current behavior

This is weird one. I don't have a repo at this time and I doubt I will be able to create one, since the case is so weird that I suspect it would involved multiple 'obscure' conditions to trigger.

I'm tring to describe the case:
- I have a few `CheckBox `instances on a `StackPanel` which ultimately is hosted by a **Uno.UI.Toolkit.ElevatedView**. Basically I'm simulating a mode dialog (I don't think that matters here, but anyway...)
- as I remove the `StackPanel `from the visual tree I will see that `.Unchecked` for the first `CheckBox` child on the `StackPanel` is fired (I confirmed by tracing)
- the `CheckBox `is not bound to anything but just listed the the `Children` collection. 
- it appears I only could reproduce the problem if the `CheckBox` was checked as the 'dialog' was closed. If feels like there would be some logic in place: "OK, this thing is closed. Now there is a checkbox which is/was checked. We have to uncheck it to clear things up." (pure speculation obviously)

Here is a snippet:
```
new CheckBox
{
HorizontalAlignment     = HorizontalAlignment.Center,
IsChecked                      = Common.Session.Current.IsExpertMode,
Margin                           = new Thickness(24, 8, 0, 0),
VerticalAlignment          = VerticalAlignment.Center,
Visibility                         = (Gui.MainPage.IsPortraitMode ? Visibility.Collapsed : Visibility.Visible),
}
```
.. there really is nothing extraordinary.

What's going on there? How could an event for `CheckBox `ever be fired after it's removed from the visual tree/DOM???

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

No

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #12182

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
While this solution is certainly better than what was there before, there could still be issues - e.g. if the template contains a `CheckBox` which is originally set to be checked, this will un-check it. So I agree that we need to address https://github.com/unoplatform/uno/issues/12189

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.mux.cs [hunk 1]
```diff
@@ -227,6 +227,12 @@ private void OnToggleImpl()
 		/// </summary>
 		private void OnIsCheckedChanged()
 		{
+			if (_suppressCheckedChanged)
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.mux.cs:230]: Add the `// Uno workaround` comment here too please

**Comment by Youssef1313** [comment]
> While this solution is certainly better than what was there before, there could still be issues - e.g. if the template contains a `CheckBox` which is originally set to be checked, this will un-check it. So I agree that we need to address #12189

Yes. This PR is only a workaround for this specific issue. But reusing templates in general is very problematic and I think we should try to remove it.

File: src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.mux.cs [hunk 2]
```diff
@@ -227,6 +227,12 @@ private void OnToggleImpl()
 		/// </summary>
 		private void OnIsCheckedChanged()
 		{
+			if (_suppressCheckedChanged) // Uno workaround
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.mux.cs:230]: Sorry, didn't realize at first, but could you also add the pooling issue link / #id to this and the other workaround comments? Just in case we are not able to address this right now, so we later know which parts of code can be removed

File: src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.cs [hunk 3]
```diff
@@ -8,6 +8,12 @@ namespace Windows.UI.Xaml.Controls.Primitives
 	/// </summary>
 	public partial class ToggleButton : ButtonBase, IFrameworkTemplatePoolAware
 	{
+		/// <summary>
+		/// This is a workaround for template pooling issue where we change IsChecked when template is recycled.
+		/// This prevents incorrect event raising, but is not a "real" solution. Pooling could still cause issues.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.cs:13]:
```suggestion
		/// This prevents incorrect event raising, but is not a "real" solution. Pooling could still cause issues.
		/// This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
```

File: src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.cs [hunk 4]
```diff
@@ -82,6 +88,11 @@ public bool IsThreeState
 
 		protected virtual void OnIsCheckedChanged(bool? oldValue, bool? newValue)
 		{
+			if (_suppressCheckedChanged)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.cs:91]:
```suggestion
			/// This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
			if (_suppressCheckedChanged)
```

File: src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.mux.cs (hunk 2 above)
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/ToggleButton.mux.cs:230]:
```suggestion
			/// This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
			if (_suppressCheckedChanged) // Uno workaround
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs [hunk 5]
```diff
@@ -43,6 +43,12 @@ public class TextBoxConstants
 
 	public partial class TextBox : Control, IFrameworkTemplatePoolAware
 	{
+		/// <summary>
+		/// This is a workaround for template pooling issue where we change IsChecked when template is recycled.
+		/// This prevents incorrect event raising, but is not a "real" solution. Pooling could still cause issues.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs:48]:
```suggestion
		/// This prevents incorrect event raising, but is not a "real" solution. Pooling could still cause issues.
		/// This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
```

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs [hunk 6]
```diff
@@ -304,13 +310,17 @@ private void RaiseTextChanged()
 			{
 				_isInvokingTextChanged = true;
 				_isTextChangedPending = false;
-				TextChanged?.Invoke(this, new TextChangedEventArgs(this));
+				if (!_suppressTextChanged)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs:313]:
```suggestion
				if (!_suppressTextChanged) // This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
```

File: src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.cs [hunk 7]
```diff
@@ -13,6 +13,12 @@ namespace Windows.UI.Xaml.Controls
 	/// </summary>
 	public partial class ToggleSwitch : Control, IFrameworkTemplatePoolAware
 	{
+		/// <summary>
+		/// This is a workaround for template pooling issue where we change IsOn when template is recycled.
+		/// This prevents incorrect event raising, but is not a "real" solution. Pooling could still cause issues.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.cs:18]:
```suggestion
		/// This prevents incorrect event raising, but is not a "real" solution. Pooling could still cause issues.
		/// This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
```

File: src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs [hunk 8]
```diff
@@ -419,7 +419,10 @@ protected virtual void OnToggled()
 			var args = new RoutedEventArgs();
 			args.OriginalSource = this;
 
-			Toggled?.Invoke(this, args);
+			if (!_suppressToggled) // Uno workaround.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs:422]:
```suggestion
			// This workaround can be removed if pooling is removed. See https://github.com/unoplatform/uno/issues/12189
			if (!_suppressToggled) // Uno workaround.
```

File: src/SamplesApp/SamplesApp.Shared/App.xaml.cs [hunk 9]
```diff
@@ -522,11 +522,6 @@ static void ConfigureFeatureFlags()
 #if __SKIA__
 			Uno.UI.FeatureConfiguration.ToolTip.UseToolTips = true;
 #endif
-
-#if HAS_UNO
-			// Allow template pool to work under higher memory load for CI.
-			FrameworkTemplatePool.HighMemoryThreshold = 0.9f;
-#endif
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.Shared/App.xaml.cs:529]: Why did you remove this one?

Comment (Youssef1313) [reply src/SamplesApp/SamplesApp.Shared/App.xaml.cs:529]: @jeromelaban It was added as a way to allow the pooling to work in CI when asked. The `MockProvider` I added achieves the exact same, but in a more reliable way.

The `0.9f` worked reliably for CI, but it doesn't always work locally.

Comment (Youssef1313) [reply src/SamplesApp/SamplesApp.Shared/App.xaml.cs:529]: For reference, https://github.com/unoplatform/uno/pull/10725

**Review by jeromelaban — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]
